src/views/PDFView.vue

```diff
@@ -33,9 +33,32 @@ export default {
 	computed: {
 		iframeSrc() {
 			return generateUrl('/apps/files_pdfviewer/?file={file}', {
-				file: this.davPath,
+				file: this.encodedDavPath,
 			})
 		},
+
+		encodedDavPath() {
+			const hasScheme = this.davPath.indexOf('://') !== -1
+
+			const pathSections = this.davPath.split('/')
+
+			// Ignore scheme and domain in the loop (note that the scheme
+			// delimiter, "//", creates an empty section when split by "/").
+			const initialSection = hasScheme ? 3 : 0
+
+			let encodedDavPath = ''
+			for (let i = initialSection; i < pathSections.length; i++) {
+				if (pathSections[i] !== '') {
+					encodedDavPath += '/' + encodeURIComponent(pathSections[i])
+				}
+			}
+
+			if (hasScheme) {
+				encodedDavPath = pathSections[0] + '//' + pathSections[2] + encodedDavPath
+			}
+
+			return encodedDavPath
+		},
```

Comment:
The question is: shouldn't this be on Viewer? we have the same issue for other files

Reply:
Yes, that question is in the pull request description...